exchange/exchange.go

```diff
@@ -34,6 +34,22 @@ type bidResponseWrapper struct {
 	bidder openrtb_ext.BidderName
 }
 
+// Container for storing a pointer to the winning bid.
+type targetData struct {
+	targetFlag bool
+	lengthMax int
+	priceGranularity openrtb_ext.PriceGranularity
+	bid map[string]*openrtb.Bid
+	bidder map[string]openrtb_ext.BidderName
```

Comment:
Should leave a comment here describing `bid` and `bidder`. ...may want to rename to something like `winningBids` and `winningBidders` too.

I spent a pretty stupidly long time to sift through the code and convince myself that the targeting keys were only being applied to _winning_ bids, and not _all_ bids. One of the big reasons is because I didn't know what these variables were doing.

It's also definitely worth commenting that the map keys are the `imp.id` values in the auction. Since the `string` type doesn't describe it clearly, the code comments need to.